CUDADataFormats/Track/interface/TrackSoAHeterogeneousDevice.h

```diff
@@ -0,0 +1,33 @@
+#ifndef CUDADataFormats_Track_TrackHeterogeneousDevice_H
+#define CUDADataFormats_Track_TrackHeterogeneousDevice_H
+
+#include <cstdint>
+
+#include "CUDADataFormats/Track/interface/PixelTrackUtilities.h"
+#include "CUDADataFormats/Common/interface/PortableDeviceCollection.h"
+
+#include "Geometry/CommonTopologies/interface/SimplePixelTopology.h"
+
+template <typename TrackerTraits>
+class TrackSoAHeterogeneousDevice : public cms::cuda::PortableDeviceCollection<TrackLayout<TrackerTraits>> {
```

Comment:
For the next round (CUDA-to-Alpaka?) I'd suggest to drop the "Heterogeneous" from the class names. At least to me the "Heterogeneous" doesn't really tell anything about the class, since it is explicitly on the device memory space.